src/api-endpoints.ts

```diff
@@ -26,13 +26,69 @@ import {
   ParentPageInput,
   PropertySchema,
   RichTextInput,
+  UpdateBlock,
 } from "./api-types"
 
 // TODO: type assertions to verify that each interface is synchronized to the list of keys in the runtime value below.
 
 // TODO: instead of importing interfaces like BlockBase, should i use a type alias to Block?
 // TODO: need an input version of Block
 
+/*
+ * blocks.retrieve()
+ */
+
+interface BlocksRetrievePathParameters {
+  block_id: string
+}
+interface BlocksRetrieveQueryParameters {}
+interface BlocksRetrieveBodyParameters {}
+
+export interface BlocksRetrieveParameters
+  extends BlocksRetrievePathParameters,
+    BlocksRetrieveQueryParameters,
+    BlocksRetrieveBodyParameters {}
+export interface BlocksRetrieveResponse extends BlockBase {}
+
+export const blocksRetrieve = {
+  method: "get",
+  pathParams: ["block_id"],
+  queryParams: [],
+  bodyParams: [],
+  path: (p: BlocksRetrievePathParameters) => `blocks/${p.block_id}`,
+} as const
+
+/*
+ * blocks.update()
+ */
+
+interface BlocksUpdatePathParameters {
+  block_id: string
+}
+interface BlocksUpdateQueryParameters {}
+type BlocksUpdateBodyParameters = UpdateBlock
+
+export type BlocksUpdateParameters = BlocksUpdatePathParameters &
+  BlocksUpdateQueryParameters &
+  BlocksUpdateBodyParameters
```

Comment:
why are these intersected?

Reply:
The `Parameters` for all the other endpoints are interfaces that extend each other so that the final type is the key/value pairs of all the params

For update block this doesn't work since the body param type uses `|`, which isn't supported for interfaces so I used `&` with types instead